File: VB/HowToLimitAmountOfObjectsCreatedViaNewAction.Module/MyController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.Persistent.BaseImpl

Namespace LimitObjectsCreatedViaNewAction.Module
	Partial Public Class MyController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
		End Sub

		Private Sub MyController_FrameAssigned(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.FrameAssigned
			Dim standardController As NewObjectViewController = Frame.GetController(Of NewObjectViewController)()
			AddHandler standardController.ObjectCreating, AddressOf standardController_ObjectCreating
		End Sub

		Private Sub standardController_ObjectCreating(ByVal sender As Object, ByVal e As ObjectCreatingEventArgs)
			If View.ObjectTypeInfo.Type Is GetType(Task) Then
				Dim count As Integer = 0
				If TypeOf View Is DevExpress.ExpressApp.ListView Then
					count = (CType(View, DevExpress.ExpressApp.ListView)).CollectionSource.GetCount()
				Else
					count = View.ObjectSpace.GetObjectsCount(GetType(Task), Nothing)
				End If
				If count >= 3 Then
					e.Cancel = True
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/HowToLimitAmountOfObjectsCreatedViaNewAction.Module/MyController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.SystemModule;
using DevExpress.Persistent.BaseImpl;

namespace HowToLimitAmountOfObjectsCreatedViaNewAction.Module {
	public partial class MyController : ViewController {
		public MyController() {
			InitializeComponent();
			RegisterActions(components);
		}

		private void MyController_FrameAssigned(object sender, EventArgs e) {
			NewObjectViewController standardController =
			Frame.GetController<NewObjectViewController>();
			standardController.ObjectCreating += new EventHandler<ObjectCreatingEventArgs>(standardController_ObjectCreating);
		}

		void standardController_ObjectCreating(object sender, ObjectCreatingEventArgs e) {
			int count = 0;
			if(View is DevExpress.ExpressApp.ListView) {
				count = ((DevExpress.ExpressApp.ListView)View).CollectionSource.GetCount();
			}
			else {
				count = View.ObjectSpace.GetObjectsCount(typeof(Task), null);
			}
			if(count >= 3) {
				e.Cancel = true;
			}
		}
	}
}
